### DISCUSSION TITLE
Change NetworkConnection.Scenes to use scene handles

### DISCUSSION THREAD

**Discussion by celojevic** [FirstGearGames/FishNet#729] [Feature Request]
This is a breaking change. I've done some testing with this and it requires just a few changes in the source to just use `scene.handle` instead of Scene. It completely removes GC and saves a little performance. Here's what it looks like before the change:
![image](https://github.com/user-attachments/assets/f1167f20-ff77-4967-aca8-853d02d37e6d)
After:
![image](https://github.com/user-attachments/assets/ab530c05-f24e-4712-809a-f684ce34faa7)

People can still use the Unity SceneManager to get scenes by their handle.

Unity 2022 LTS
FN 3.11.7

**Comment by FirstGearGames** [comment]
Before we go this route have you tried using an equality comparer on the
hash
set which uses the scene Id?

**Reply by celojevic** [reply]
I just tried it, same performance gain, thanks for the idea :D

**Comment by FirstGearGames** [comment]
Since you already tested are you able to submit a PR? If you don't want to
just post your changes here please.

**Reply by celojevic** [reply]
i just gave it a public setter and made a new HashSet in a custom script with Server.OnRemoteConnectionState since thats the only way to set a comparer
```
        private void ServerManager_OnRemoteConnectionState(FishNet.Connection.NetworkConnection conn, FishNet.Transporting.RemoteConnectionStateArgs arg2)
        {
            if (arg2.ConnectionState == FishNet.Transporting.RemoteConnectionState.Started)
            {
                conn.Scenes = new HashSet<Scene>(_sceneComparer);
            }
        }
    }
    public class SceneEqualityComparer : IEqualityComparer<Scene>
    {
        public bool Equals(Scene x, Scene y)
        {
            return x.handle == y.handle;
        }

        public int GetHashCode(Scene obj)
        {
            return obj.GetHashCode();
        }
    }```
